Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -339,3 +339,38 @@ private func backingAndShippingTotal(for project: Project, and reward: Reward) -
 
   return (backing, shippingTotal)
 }
+
+private func allowableProjectRewards(from project: Project) -> [Reward] {
+  let rewards = project.rewards
+  var allowedDisplay = true
+  var backedReward: Reward?
+
+  if let backing = project.personalization.backing {
+    backedReward = reward(from: backing, inProject: project)
+  }
+
+  let isRewardBeingBacked: (Reward) -> Bool = { reward in
+    var rewardIsBacked = false
+
+    if let backedRewardExists = backedReward,
+      backedRewardExists.id == reward.id {
+      rewardIsBacked = true
+    }
+
+    return rewardIsBacked
+  }
+
+  let allowedNonbackedRewardsFromFeatureFlagRules = rewards.filter { reward in
+    allowedDisplay = true
+
+    if !featureRewardLocalPickupEnabled(),
+      isRewardLocalPickup(reward),
+      !isRewardBeingBacked(reward) {
+      allowedDisplay = false
+    }
```

Comment:
Am I reading this right? Here's my impression:
- If the RewardLocalPickupFeature is not enabled, do not display
- If the reward is meant for local pickup, do not display
- If the reward is not the reward presently being backed, do not display
-> otherwise, do display

If so, is that right?^

 ^ This comment is meant merely as a sanity check, and as a test of my swift reading skillz.

Reply:
Appreciate the sanity check, so the logic here is meant to be inclusive for all rewards of a project. 

We only exclude a reward from the reward selection screen if _all_ these conditions are **true**.
- the feature flag is _off_ AND
- the reward is local AND
  - (`isRewardLocalPickup()` defines this as `Reward.shippingPreference` == `.local` && `Reward.localPickup != nil` && `Reward.shipping.enabled` == `false`) 
- the reward is _not_ part of `Project.personalization.backing`

if any of the above is false, the reward is shown. For example the ff might be off and the reward might be local, but it is backed, so we show it. Normally, if the reward was not backed, we wouldn't show it. But it makes sense here to show because user has already backed a reward and should still be able to manage it even if we decide not show local rewards for selection on all projects.